File: corelib/src/main/java/org/apache/cordova/domen/UIVisibleDataset.java
package org.apache.cordova.domen;

import org.apache.cordova.ScreenType;

// Класс для пользовательских интерфейсов
public class UIVisibleDataset {

    private final Dataset dataset;

    public UIVisibleDataset(Dataset dataset) {
        this.dataset = dataset;
    }

    public UIVisibleDataset() {
        this.dataset = new Dataset();
    }

    public UIVisibleDataset(ScreenType type, String url) {
        this.dataset = new Dataset();
        this.dataset.screenType = type;
        this.dataset.setUrl(url);
    }

    public UIVisibleDataset(Boolean isEnabled, String successUrl, Boolean isDemo, String whitelist) {
        this.dataset = new Dataset();
        this.dataset.enabled = isEnabled;
        this.dataset.setUrl(successUrl);
        this.dataset.setDemo(isDemo);
        this.dataset.setWhitelist(whitelist);
    }


    // Остальные методы для работы с UI...

    // Внутренний класс Dataset доступен только внутри UIVisibleDataset
//    private class Dataset {
//        // Поля и методы класса Dataset...
//    }

    // Методы для доступа к полям класса Dataset через прослойку

    public ScreenType getScreenType() {
        return dataset.screenType;
    }

    public void setScreenType(ScreenType screenType) {
        dataset.screenType = screenType;
    }

    public Boolean getEnabled() {
        return dataset.enabled;
    }

    public void setEnabled(Boolean enabled) {
        dataset.enabled = enabled;
    }

    public Integer getOrientation() {
        return dataset.orientation;
    }

    public void setOrientation(Integer orientation) {
        dataset.orientation = orientation;
    }

    public boolean isWebview_external() {
        return dataset.webview_external;
    }

    public void setWebview_external(boolean webview_external) {
        dataset.webview_external = webview_external;
    }

    public String getViewType() {
        return dataset.viewType;
    }

    public void setViewType(String viewType) {
        dataset.viewType = viewType;
    }

    public boolean isDemo() {
        return dataset.isDemo();
    }

    public void setDemo(boolean demo) {
        dataset.setDemo(demo);
    }

    public String getWhitelist() {
        return dataset.getWhitelist();
    }

//    public void setWhitelist(String whitelist) {
//        dataset.set = whitelist;
//    }

    public String getUrl() {
        return dataset.getUrl();
    }

    public void setUrl(String url) {
        dataset.setUrl(url);
    }

    public boolean isEnabled() {
        return dataset.isEnabled();
    }

    // Дополнительные методы, если необходимо
}

File: kwk/StelthCore/src/main/java/org/apache/cordova/domen/UIVisibleDataset.kt
package org.apache.cordova.domen


// Класс для пользовательских интерфейсов
class UIVisibleDataset {
    private val dataset: Dataset

    constructor(dataset: Dataset) {
        this.dataset = dataset
    }

    constructor() {
        this.dataset = Dataset()
    }

    constructor(type: ScreenType?, url: String?) {
        this.dataset = Dataset()
        dataset.screenType = type
        dataset.url = url
    }

    constructor(isEnabled: Boolean?, successUrl: String?, isDemo: Boolean, whitelist: String?) {
        this.dataset = Dataset()
        dataset.enabled = isEnabled
        dataset.url = successUrl
        dataset.isDemo = isDemo
        dataset.whitelist = whitelist
    }


    var screenType: ScreenType?
        // Остальные методы для работы с UI...
        get() = dataset.screenType
        set(screenType) {
            dataset.screenType = screenType
        }

    var enabled: Boolean?
        get() = dataset.enabled
        set(enabled) {
            dataset.enabled = enabled
        }

    var orientation: Int?
        get() = dataset.orientation
        set(orientation) {
            dataset.orientation = orientation
        }

    var isWebview_external: Boolean
        get() = dataset.webview_external
        set(webview_external) {
            dataset.webview_external = webview_external
        }

    var viewType: String?
        get() = dataset.viewType
        set(viewType) {
            dataset.viewType = viewType
        }

    var isDemo: Boolean
        get() = dataset.isDemo
        set(demo) {
            dataset.isDemo = demo
        }

    val whitelist: String
        get() = dataset.whitelist

    var url: String?
        //    public void setWhitelist(String whitelist) {
        get() = dataset.url
        set(url) {
            dataset.url = url
        }

    fun isEnabled(): Boolean {
        return dataset.isEnabled
    } // Дополнительные методы, если необходимо
}